### PR TITLE
Added setGameTile/getGameTitle and setGameIcon/getGameIcon

### PR DESCRIPTION
**Pull request by FriskTheFallenHuman**
This PR allows mods to change the Game's Title bar and Icon from the `mod.lua` file

### CODE REVIEW

File: src/engine/game/game.lua [hunk 1]
```diff
@@ -542,6 +542,26 @@ function Game:getSoulPartyMember()
...
+    love.window.setTitle(title)
+end
+
+function Game:getGameTitle()
+    return love.window.getTitle()
+end
+
+function Game:setGameIcon(iconData)
+    -- Make sure our icon isn't null.
+    if iconData == nil then return end
```
Comment (NyakoFox) [inline src/engine/game/game.lua:555]: i dont know if this is required, if someone passes bad data then it _should_ error

File: src/engine/vars.lua [hunk 2]
```diff
@@ -1,11 +1,13 @@
  -- replaces mod list with "Start game", should be a string of the mod id
 TARGET_MOD = nil
-
+TARGET_ICON = "icon.png"
```
Comment (NyakoFox) [inline src/engine/vars.lua:3]: why did you add unused variables here?

File: src/engine/game/game.lua [hunk 3]
```diff
@@ -542,6 +542,26 @@ function Game:getSoulPartyMember()
     return current
 end
 
+function Game:setGameTitle(title)
```
Comment (NyakoFox) [inline src/engine/game/game.lua:545]: maybe its a good idea to change "Game" to "Window" in these functions? ex. `setWindowTitle()`

File: mods/_testmod/mod.lua [hunk 4]
```diff
@@ -19,6 +19,13 @@ function Mod:init()
 end
 
 function Mod:preInit()
+    -- Set the game title to something different
+    Game:setGameTitle("Example Mod")
+
+    -- Set the windows icon to something different
+    -- For mods please use Assets.getTextureData has argument
+    Game:setGameIcon( Assets.getTextureData( "kristal/banana_1" ) )
```
Comment (NyakoFox) [inline mods/_testmod/mod.lua:27]: This isn't where `Assets.getTextureData` should be, it should be handled in the function itself

**Review by NyakoFox — CHANGES_REQUESTED** [review]

**Comment by DimitriBarronmore** [comment]
I feel like I have to point out the obvious here and say that this isn't necessary. The only one of these functions which isn't a single line arguably should have been, and nothing stops the modder from writing those single lines themselves.

Instead of adding pointless aliases which could essentially be replaced with a variable pointing to the original function, a more useful step towards making it simple for modders to change the icon and title would just be to make sure they reset when the game transitions back to the menu. This doesn't solve that or any problems, it just makes things more confusing.
